src/store/token-dashboard/slice.ts

```diff
@@ -150,11 +172,20 @@ const slice = createSlice({
     setAssociatedWallets: (
       state,
       {
-        payload: { associatedWallets }
-      }: PayloadAction<{ associatedWallets: AssociatedWallets }>
+        payload: { associatedWallets, chain }
+      }: PayloadAction<{ associatedWallets: AssociatedWallets; chain: Chain }>
     ) => {
-      state.associatedWallets.connectedWallets = associatedWallets
-      state.associatedWallets.confirmingWallet = null
+      if (chain === 'sol') {
+        state.associatedWallets.connectedSolWallets = associatedWallets
```

Comment:
Why have separate collections of wallets rather than one collection of different wallet types (eg a "chain" prop on the wallet themselves)? It seems like a lot of the work here is in splitting the logic between the two collections.

Reply:
Yeah, I think it'd be the same amount of work either way.
The Solana wallets and Eth wallets are treated differently: Fetching their balances, signing messages, and formatting/displaying the wallets in the UI differentiates the two.
I could go either way for how it's stored though, just thought it was more explicit to split them up this way so we don't mix methods between the two.